src/vm/methodtable.cpp

```diff
@@ -1713,6 +1717,13 @@ TypeHandle::CastResult MethodTable::CanCastToClassNoGC(MethodTable *pTargetMT)
     }
     CONTRACTL_END
 
+    // allow an object of type T to be cast to Nullable<T> (they have the same representation)
```

Comment:
This change does not look right. `T` -> `Nullable<T>` cast is valid only if the source is boxed. I am surprised that there are no tests failing because of this change.

Reply:
Currently unboxed types are a special case and require user to call special helpers. 
I assume that is because it is fairly rare that you deal with unboxed types - generally when reasoning about array elements and such. 

We have comments like 
```
    // Note that if the TypeHandle is a valuetype, the caller is responsible
    // for checking that the valuetype is in its boxed form before calling
    // CanCastTo. Otherwise, the caller should be using IsBoxedAndCanCastTo()
```

Moving nullable check together with the comment here is basically a refactoring. It was done in the same way before, just in several callers and in different order with respect to calling this one. Moving it here makes more uniform and allows to know when we are at the end of all checks and can cache the result.

We have a lot of stuff that relies on castability of nullables in CoreFX tests like Linq and Expressions. Mixing up nullable conversions even in subtle ways invariably resulted in numerous test failures, so coverage for nullables is pretty decent.